crates/starknet-devnet/tests/test_get_events.rs

```diff
@@ -151,4 +151,44 @@ mod get_events_integration_tests {
 
         get_events_correct_chunking(&devnet, true).await
     }
+
+    #[tokio::test]
+    async fn get_events_errors() {
+        let devnet = BackgroundDevnet::spawn().await.expect("Could not start Devnet");
+
+        let chunk_size = 3;
+        let mut continuation_token: Option<String> = None;
+        let event_filter = EventFilter {
+            from_block: Some(BlockId::Number(90000000)),
+            to_block: Some(BlockId::Tag(BlockTag::Latest)),
+            address: None,
+            keys: None,
+        };
+        match devnet
+            .json_rpc_client
+            .get_events(event_filter.clone(), continuation_token.clone(), chunk_size as u64)
+            .await
+            .unwrap_err()
+        {
+            ProviderError::StarknetError(StarknetError::BlockNotFound) => (),
+            err => panic!("Invalid error: {err:?}"),
+        }
+
+        continuation_token = Some(String::from("invalid token"));
+        let event_filter = EventFilter {
+            from_block: Some(BlockId::Number(0)),
+            to_block: Some(BlockId::Tag(BlockTag::Latest)),
+            address: None,
+            keys: None,
+        };
+        match devnet
+            .json_rpc_client
+            .get_events(event_filter.clone(), continuation_token.clone(), chunk_size as u64)
```

Comment:
Is `clone` necessary here? The variables are not used after this line anymore. But in fact, instead of redefining variables or making them mutable, why not just use two blocks with their own namespaces? E.g. L177-L193 could be put inside of curly braces.

Reply:
you right, it would be much more clear